eos/utils/integrate.hh

```diff
@@ -43,6 +44,69 @@ namespace eos
 
     template <std::size_t k> std::array<double, k> integrate1D(const std::function<std::array<double, k> (const double &)> & f, unsigned n, const double & a, const double & b);
     /// @}
+
+namespace GSL
+{
+    using fdd = std::function<double(const double &)>;
+    struct QNG
+    {
+        class Config
+        {
+            public:
+                Config();
+
+
+
+                double epsabs() const;
+                Config& epsabs(const double& x);
+
+                double epsrel() const;
+                Config& epsrel(const double& x);
+            private:
+              double _epsabs, _epsrel;
+        };
+    };
+
+    struct QAGS
+    {
+        class Config : public QNG::Config
+        {
+            public:
+                Config();
+
+                size_t limit() const;
+                Config& limit(const size_t&);
+
+                int key() const;
+                Config& key(const int&);
```

Comment:
I think the ```key``` should be known at compile time. An elegant solution would be to use:
``` c++
namespace impl { class QAGSConfig { /* without the key */... }; }
template <unsigned points_> struct QAGS
{
    using Config = impl::QAGSConfig;
    static const key;
};
static const QAGS<15>::key = GSL_INTEG_GAUSS15;
...
static const QAGS<61>::key = GSL_INTEG_GAUSS61;
```

What do you think?

Reply:
That would work. Or with an enum

Reply:
Enum is also fine, yes.